### PR TITLE
CRI-O: fix the storage ref when copying images

### PR DESCRIPTION
**Pull request by runcom**
Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
@kargakis PTAL

File: sjb/config/test_cases/test_branch_origin_extended_conformance_crio.yml [hunk 1]
```diff
@@ -87,7 +87,7 @@ extensions:
       timeout: 10800
       script: |-
         for i in $(docker images --format '{{.Repository}}:{{.Tag}}' | grep "openshift\/origin"); do
-          sudo skopeo copy docker-daemon:$i containers-storage:$i
+          sudo skopeo copy docker-daemon:$i containers-storage:[overlay2@/var/lib/containers/storage+/var/run/containers/storage:overlay2.override_kernel_check=1]$i
```
Comment (runcom) [inline sjb/config/test_cases/test_branch_origin_extended_conformance_crio.yml]: do I have to quote brackets here? cause I have to in my shell but I'm not sure about it here @kargakis @stevekuznetsov

Comment (kargakis) [reply sjb/config/test_cases/test_branch_origin_extended_conformance_crio.yml]: Yeah, you probably need to escape these. Not 100% sure though but we can try it.

Comment (runcom) [reply sjb/config/test_cases/test_branch_origin_extended_conformance_crio.yml]: so I'll repush with those escaped? or we can try this commit first?

Comment (runcom) [reply sjb/config/test_cases/test_branch_origin_extended_conformance_crio.yml]: alright I've re-pushed with brackets escaped

**Comment by kargakis** [comment]
Testing at https://ci.openshift.redhat.com/jenkins/view/All/job/test_branch_origin_extended_conformance_crio/21/

**Comment by runcom** [comment]
@kargakis can you stop build 21 and re-kick? I've just fixed another thing

**Comment by runcom** [comment]
btw, there seems to be some network issue:
```
2017-09-14T09:06:50.503866577-04:00 I0914 13:06:50.503847       1 source.go:167] fatal: unable to access 'https://github.com/sclorg/s2i-php-container/': Could not resolve host: github.com; Unknown error
2017-09-14T09:06:50.519948389-04:00 F0914 13:06:50.519918       1 helpers.go:120] error: fatal: unable to access 'https://github.com/sclorg/s2i-php-container/': Could not resolve host: github.com; Unknown error
```
@kargakis is this a known issue?

**Comment by runcom** [comment]
@kargakis flacky again though:
```
[INFO] [CLEANUP] Pruning etcd data directory
rm: cannot remove ‘/tmp/etcd/member’: Permission denied
```
can you re-kick a job?

**Comment by stevekuznetsov** [comment]
@runcom 
> rm: cannot remove ‘/tmp/etcd/member’: Permission denied

this is a red herring

**Comment by stevekuznetsov** [comment]
We need to enable jUnit publishing for the job:

```
FAIL! -- 227 Passed | 20 Failed | 5 Pending | 553 Skipped 
```

**Comment by runcom** [comment]
> We need to enable jUnit publishing for the job:
> 
> FAIL! -- 227 Passed | 20 Failed | 5 Pending | 553 Skipped

yup, let me figure the situation before, then we can enable junit

**Comment by runcom** [comment]
triggered a new job myself :tada:

**Comment by kargakis** [comment]
:tada: :tada: :tada:

**Comment by kargakis** [comment]
@runcom rename the PR when you are ready

**Comment by runcom** [comment]
> @runcom rename the PR when you are ready

yeah, I noticed `openshift-deploy` pod isn't running so maybe it's selinux and I have to test this out, I'll rename the PR once ready.

**Comment by runcom** [comment]
@kargakis @stevekuznetsov we need logs from `/var/log/pods/*` but the CI isn't collecting them, is there any fast way to collect them?

**Comment by runcom** [comment]
> @kargakis @stevekuznetsov we need logs from /var/log/pods/* but the CI isn't collecting them, is there any fast way to collect them?

pushed a commit to collect that dir thx @stevekuznetsov

**Comment by runcom** [comment]
@kargakis @stevekuznetsov @smarterclayton the issue with the Origin CI and CRI-O is the following:
```
2017-09-15T11:54:38.089860076-04:00 stderr /usr/bin/openshift-deploy: error while loading shared libraries: libpthread.so.0: cannot open shared object file: Permission denied
```
I doubt it's related to CRI-O, could you guys look into this?

**Comment by smarterclayton** [comment]
Hrm - I've never seen that on Docker.  Is there any chance that LD_PRELOAD or LD_LIBRARY_PATH are either being set as env vars unintentionally in the container?

**Comment by runcom** [comment]
Does anyone know why CI keeps failing https://ci.openshift.redhat.com/jenkins/job/test_branch_origin_extended_conformance_crio/39/ ? It's a permission denied on etcd stuff, @kargakis @stevekuznetsov PTAL

**Comment by stevekuznetsov** [comment]
Error is:

```

  1. Hosts:    localhost
     Play:     Configure masters
     Task:     Start and enable master controller on first master
     Message:  Unable to start service origin-master-controllers: Job for origin-master-controllers.service failed because the control process exited with error code. See "systemctl status origin-master-controllers.service" and "journalctl -xe" for details.
```

**Comment by runcom** [comment]
Yeah sorry, that one. From logs in can't see anything related to cri-o itself so was just wondering if that error is known.

**Comment by stevekuznetsov** [comment]
Installer issue?

```
Sep 17 06:27:44 ip-172-18-8-157.ec2.internal systemd[1]: Starting Atomic OpenShift Master Controllers...
Sep 17 06:27:44 ip-172-18-8-157.ec2.internal origin-master-controllers[17307]: F0917 06:27:44.673616   17307 start_controllers.go:66] could not load config file "/etc/origin/master/master-config.yaml" due to an error: error reading config: [pos 2375]: json: expect char '{' but got char '"'
Sep 17 06:27:44 ip-172-18-8-157.ec2.internal systemd[1]: origin-master-controllers.service: main process exited, code=exited, status=255/n/a
Sep 17 06:27:44 ip-172-18-8-157.ec2.internal systemd[1]: Failed to start Atomic OpenShift Master Controllers.
Sep 17 06:27:44 ip-172-18-8-157.ec2.internal systemd[1]: Unit origin-master-controllers.service entered failed state.
Sep 17 06:27:44 ip-172-18-8-157.ec2.internal systemd[1]: origin-master-controllers.service failed.
```

**Comment by stevekuznetsov** [comment]
I've never seen that before

**Comment by stevekuznetsov** [comment]
Could add a debug cat of the master config as a post step in this PR to test

**Comment by runcom** [comment]
@stevekuznetsov so something went wrong during the weekend, you can see from the logs that the steps I've added in this PR aren't run in the linked CI job (search for `"copy openshift images from docker storage to CRI-O storage"` in console output for job #39 for instance). Now it seems it's running it...not sure what's going on.

**Comment by runcom** [comment]
@kargakis @stevekuznetsov this is good to go now, please merge it

**Comment by kargakis** [comment]
@runcom I guess you have already pushed in Jenkins?

File: sjb/config/test_cases/test_branch_origin_extended_conformance_crio.yml [hunk 2]
```diff
@@ -46,11 +46,21 @@ extensions:
...
+          sudo skopeo copy docker-daemon:$i containers-storage:\[overlay2@/var/lib/containers/storage+/var/run/containers/storage:overlay2.override_kernel_check=1\]$i
+        done
+    - type: "script"
       title: "install origin"
       timeout: 7200
       repository: "aos-cd-jobs"
       script: |-
         local_ip="$( curl http://169.254.169.254/latest/meta-data/local-ipv4 )"
+        # debug for Dan
+        sudo setenforce 0
```
Comment (runcom) [inline sjb/config/test_cases/test_branch_origin_extended_conformance_crio.yml]: I'll remove this in a follow up, need to test separately what's going on /cc @mrunalp

Comment (kargakis) [reply sjb/config/test_cases/test_branch_origin_extended_conformance_crio.yml]: Please open an issue to track this.

**Comment by kargakis** [comment]
> @runcom I guess you have already pushed in Jenkins?

Ok, updated.

**Pull request merged**